src/tests/JIT/opt/InstructionCombining/AcrossAndCselToAcross.cs

```diff
@@ -0,0 +1,77 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+// unit test for the full range comparison optimization
+
+using System;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.Intrinsics;
+using System.Runtime.Intrinsics.Arm;
+using Xunit;
+
+public class AcrossAndCselToAcross
+{
+
+    [Theory]
+    [InlineData(1, 42)]
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    public static Vector<long> addAcross_sbyte(sbyte op1, sbyte op2)
+    {
+        //ARM64-FULL-LINE: cmpne {{p[0-9]+.b}}, {{p[0-9]+/z}}, {{z[0-9]+.b}}, #0
+        //ARM64-FULL-LINE-NEXT: saddv {{d[0-9]+}}, {{p[0-9]+}}, {{z[0-9]+.b}}
+        var vec = Vector.Create(op2);
+        var mask = Vector.Create(op1);
+        return Sve.AddAcross(Sve.ConditionalSelect(mask, vec, Vector<sbyte>.Zero));
```

Comment:
My concern with these optimizations is that users will not know about them and will end up writing non-optimal code. Is there somewhere we could document these?

Reply:
yes, and there are lot of other non-sve non-arm64 scenarios where someone writes code produces sub-optimal code. The best we can do it put these things together in a blog post that we publish occasionally.